### PR TITLE
Add swipe actions to cells

### LINKED ISSUE #28: Add swiping to the time entry cell
Add the swiping gestures and UI to the cells. Bind those to send the continue and delete actions.

### PR DESCRIPTION
**Pull request by Odrakir**
Closes #28 
Supersedes #81 

Some notes:
- I'm not using `Button` for the continue and delete buttons because they clash with the cell somehow.
- The property that drives the behaviour is `percentDragged` every other property is there to support it
- I tried, but I couldn't get rid of the press animation that's default on list cells... If you have any ideas, they are welcome.

### CODE REVIEW

File: TogglWatch WatchKit Extension/Views/TimeEntry/TimeEntryCellView.swift [hunk 1]
```diff
@@ -9,67 +9,126 @@
...
+                    
+                    ProjectTaskClientTextView(timeEntry)
+                        .multilineTextAlignment(.leading)
+                        .lineLimit(3)
+                    
+                    HStack {
+                        Spacer()
+                        Text(timeEntry.durationString ?? "")
+                            .font(.system(size: 14))
+                            .foregroundColor(Color.init(red: 174/255, green: 180/255, blue: 191/255))
```
Comment (sofokli1) [inline TogglWatch WatchKit Extension/Views/TimeEntry/TimeEntryCellView.swift]: this color can go in the extension

File: TogglWatch WatchKit Extension/Views/TimeEntry/TimeEntryCellView.swift [hunk 2]
```diff
@@ -9,67 +9,126 @@
...
+public enum SwipeButtonType
+{
+    case continueTimeEntry
+    case deleteTimeEntry
+}
+
+public struct SwipeButton: View
+{
+    var type: SwipeButtonType
+    var p: CGFloat
```
[the remark this answers is not in the document]
Comment (Odrakir) [reply TogglWatch WatchKit Extension/Views/TimeEntry/TimeEntryCellView.swift]: Yep, for me is obvious enough. Do you have any other idea?

**Comment by Odrakir** [comment]
The running TEs won't appear in this list, they'll appear on top of it as it's shown in the designs.

**Comment by Odrakir** [comment]
I can't reproduce the one about going back and forth in a single cell... Any ideas how to do it?

File: TogglWatch WatchKit Extension/Views/TimeEntry/TimeEntryCellView.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (sofokli1) [reply TogglWatch WatchKit Extension/Views/TimeEntry/TimeEntryCellView.swift]: can't think of anything else other than just `percentage`

**Comment by sofokli1** [comment]
I manage to reproduce it consistently. If you open continue let go and continue to open in the same direction the delete becomes visible. Same if you do it the other way around.

**Comment by sofokli1** [comment]
sometimes the cells touch the section header
Simulator Screen Shot - Apple Watch Series 5 - 40mm - 2019-12-02 at 19 11 07

**Comment by Odrakir** [comment]
🤔 This is so weird... I'm unable to get any of these glitches. On what Watch model is this happening?

**Comment by sofokli1** [comment]
simulator series 5 40mm 6.1os

**Comment by sofokli1** [comment]
try scrolling with the crown up and down
Dec-03-2019 08-56-22

**Comment by sofokli1** [comment]
do you want to do something about this or leave it for now?
Dec-03-2019 14-42-30

**Comment by Odrakir** [comment]
Oh, I didn't understand until now how to reproduce it. I'll fix it.

**Pull request merged**